database/src/models/status.rs

```diff
@@ -75,18 +75,20 @@ impl Status {
     }
 
     /// Returns a URI to the ActivityPub object of this status.
-    pub fn get_uri<'a>(&'a self, db_conn: &Connection) -> QueryResult<Cow<'a, str>> {
-        let uri = self.uri.as_ref().map(|x| String::as_str(x).into());
-        match uri {
-            Some(x) => Ok(x),
-            None => {
-                return Ok(format!(
-                    "{base}/users/{user}/statuses/{id}",
-                    base = BASE_URL.as_str(),
-                    user = self.account(db_conn)?.username,
-                    id = self.id
-                ).into())
-            },
+    pub fn get_uri<'a>(&'a self, db_conn: &'a Connection) -> QueryResult<Cow<'a, str>> {
+        let account = self.account(db_conn)?;
+        Ok(self.status_uri(&account))
+    }
+
+    pub fn status_uri<'a>(&'a self, account: &Account) -> Cow<'a, str> {
```

Comment:
On further thought, I don't mind this; I'd just prefer it to be named something like `uri_with_account`.